src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+    authorsString += (index > 0 ? ', ' : '') + a.fullName;
+  });
+  return authorsString;
+};
+
+const getContentString = (content: string | Author[] | undefined, id: string): string => {
+  if (typeof content == 'undefined') {
+    return '';
+  } else if (typeof content != 'string' && id === 'authors') {
+    return returnFormatedAuthorsString(content);
+  } else if (id === 'citationKey') {
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => {
+  const [value, setValue] = useState(content);
+
+  return (
+    <input
+      className="w-full border bg-slate-50 px-2 py-0.5"
+      name={id}
+      type="text"
+      value={value}
+      onBlur={(evt) => {
+        onChangeHandler(id, evt.target.value);
+      }}
+      onChange={(evt) => {
+        setValue(evt.target.value);
+      }}
+    />
+  );
+};
+
+const TableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+  editable,
+}: {
+  contentData: string | Author[];
+  id: string;
+  onChangeHandler: CallableFunction;
+  editable: boolean;
+}) => {
+  const { editing } = useContext(EditingContext);
+
+  if (!editing || !editable || typeof contentData != 'string') {
+    return <UnEditableTableDataCell contentData={contentData} id={id} />;
+  } else {
+    return <EditableTableDataCell contentData={contentData} id={id} onChangeHandler={onChangeHandler} />;
+  }
+};
+
+const EditableTableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+}: {
+  contentData: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => (
+  <td className="w-auto p-5">
+    <DataTextInput content={contentData} id={id} onChangeHandler={onChangeHandler} />
+  </td>
+);
+
+const UnEditableTableDataCell = ({ contentData, id }: { contentData: string | Author[]; id: string }) => {
+  const contentString = getContentString(contentData, id);
+
+  return (
+    <td className="w-auto p-5">
+      <span className="leading-[30px]">{contentString}</span>
+    </td>
+  );
+};
+
+const TableHeadCell = ({ content, colSpan, header }: { content: string; colSpan?: number; header?: boolean }) => {
+  const colSpanString = { ['colSpan']: colSpan };
+
+  return (
+    <th
+      className={cx(
+        { 'rounded-t bg-card-bg-header p-3 text-white': header },
+        { 'w-1/6 p-6': !header },
+        'text-sm uppercase',
+      )}
+      key={content}
+      {...colSpanString}
+    >
+      {content}
+      {IconButtons(header)}
+    </th>
+  );
+};
+
+const IconButtons = (header: boolean | undefined): ReactElement[] => {
+  if (header) {
+    return [
+      IconButton({ hiddenWhenEditing: true, icon: EditIcon(), title: 'Edit Reference' }),
+      IconButton({ hiddenWhenEditing: false, icon: CloseIcon(), title: 'Finished Editing Reference' }),
+    ];
+  } else {
+    return [];
+  }
+};
```

Comment:
Same comment as above, I think you should remove `IconButtons` and write this logic in `TableHeadCell`. Also please use the JSX syntax for components instead of regular function syntax.
```typescript
...
{content}
{header &&
  (editing ?
    <IconButton icon={<CloseIcon />} key="closeIcon" title="Finished Editing Reference" onClick={() => setEditing(false)} /> :
    <IconButton icon={<EditIcon />} key="editIcon" title="Edit Reference" onClick={() => setEditing(true)} />)
}
```